quodlibet/qltk/download.py

```diff
@@ -57,22 +57,44 @@ def _downloaded(self, msg: Soup.Message, result: Any, data: tuple) -> None:
         path, song = data
         try:
             headers = msg.get_property("response-headers")
-            size = int(headers.get("content-length"))
-            content_type = headers.get("content-type")
+
+            # Taille (en octets) depuis les headers HTTP
+            try:
+                size = headers.get_content_length()
+            except AttributeError:
+                size = None
+
+            # Type MIME depuis les headers HTTP
+            try:
+                ct = (
+                    headers.get_content_type()
+                )  # retourne (mimetype, params) en libsoup3
+                content_type = ct[0] if isinstance(ct, tuple) else str(ct)
+            except AttributeError:
+                content_type = "application/octet-stream"
+
+            size_str = (
+                format_size(size) if (size is not None and size > 0) else "unknown size"
+            )
             print_d(
-                f"Downloaded {format_size(size)} of {content_type}: {song('title')}"
+                f"Downloaded {format_size(size_str)} of {content_type}: {song('title')}"
```

Comment:
This will cause an error, no? `format_size` doesn't run on a string (it's already been called by this point)